pkg/pillar/cmd/volumemgr/handledownloader.go

```diff
@@ -39,33 +38,27 @@ func AddOrRefcountDownloaderConfig(ctx *volumemgrContext, status types.ContentTr
 		// in the downloader config before publishing
 		// Same is true for other fields
 	} else {
-		log.Debugf("AddOrRefcountDownloaderConfig: add for %s",
-			status.ContentSha256)
+		log.Debugf("AddOrRefcountDownloaderConfig: add for %s", blob.Sha256)
 	}
-	name := status.RelativeURL
 
 	// where should the final downloaded file be?
-	locFilename := path.Join(types.DownloadDirname, status.ObjType, "pending", status.ContentID.String(), path.Base(name))
+	locFilename := path.Join(types.DownloadDirname, "pending", blob.Sha256)
```

Comment:
A hacky workaround would be to add blob.Sha256 twice to the pathname. Alternatively, fix the offending code in the verifier,

Reply:
This was a brilliant catch; thank you for finding it. I updated the offending code in verifier.